packages/engine/src/scene/components/AudioAnalysisComponent.ts

```diff
@@ -0,0 +1,138 @@
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import {
+  defineComponent,
+  getComponent,
+  setComponent,
+  useComponent,
+  useEntityContext,
+  useOptionalComponent
+} from '@etherealengine/ecs'
+import { TransformComponent } from '@etherealengine/spatial'
+import { NameComponent } from '@etherealengine/spatial/src/common/NameComponent'
+import { VisibleComponent } from '@etherealengine/spatial/src/renderer/components/VisibleComponent'
+import { useEffect } from 'react'
+import { Color, Mesh, MeshStandardMaterial, SphereGeometry } from 'three'
+import { PositionalAudioComponent } from '../../audio/components/PositionalAudioComponent'
+import { AudioNodeGroups, MediaComponent, MediaElementComponent } from './MediaComponent'
+
+export const AudioAnalysisComponent = defineComponent({
+  name: 'EE_audioAnalyzer',
+  jsonID: 'audio-analyzer',
+
+  onInit: (entity) => {
+    return {
+      src: '',
+      dataArray: null as Uint8Array | null,
+      analyser: null as AnalyserNode | null,
+      bassEnabled: true as boolean,
+      midEnabled: true as boolean,
+      trebleEnabled: true as boolean,
+      bassMultiplier: 1 as number,
+      midMultiplier: 1 as number,
+      trebleMultiplier: 1 as number
+    }
+  },
+
+  onSet: (entity, component, json) => {
+    if (!json) return
+    if (json.src && component.src.value !== json.src) {
+      component.src.set(json.src)
+    }
+    if (json.dataArray && component.dataArray.value !== json.dataArray) {
+      component.dataArray.set(json.dataArray)
+    }
+    if (json.bassEnabled) {
+      component.bassEnabled.set(json.bassEnabled)
+    }
+    if (json.midEnabled) {
+      component.midEnabled.set(json.midEnabled)
+    }
+    if (json.trebleEnabled) {
+      component.trebleEnabled.set(json.trebleEnabled)
+    }
+    if (json.bassMultiplier) {
+      component.bassMultiplier.set(json.bassMultiplier)
+    }
+    if (json.midMultiplier) {
+      component.midMultiplier.set(json.midMultiplier)
+    }
+    if (json.trebleMultiplier) {
+      component.trebleMultiplier.set(json.trebleMultiplier)
+    }
+  },
+
+  toJSON: (entity, component) => {
+    return {
+      src: component.src.value,
+      dataArray: component.dataArray.value,
+      bassEnabled: component.bassEnabled.value,
+      midEnabled: component.midEnabled.value,
+      trebleEnabled: component.trebleEnabled.value,
+      bassMultiplier: component.bassMultiplier.value,
+      midMultiplier: component.midMultiplier.value,
+      trebleMultiplier: component.trebleMultiplier.value
+    }
+  },
+
+  reactor: () => {
+    const entity = useEntityContext()
+    const audioAnaylsisComponent = useComponent(entity, AudioAnalysisComponent)
+    const posAudio = useOptionalComponent(entity, PositionalAudioComponent)
+    const existingSystem = useComponent(entity, MediaComponent)
+
+    useEffect(() => {
+      setComponent(entity, VisibleComponent)
```

Comment:
scene data like `VisibleComponent`, `NameComponent`, `TransformComponent` should be set from the scene JSON rather than set here